tests/ui/infinite/infinite-vec-type-recursion.feature.stderr

```diff
@@ -0,0 +1,11 @@
+error[E0275]: overflow evaluating the requirement `X`
+  --> $DIR/infinite-vec-type-recursion.rs:9:20
+   |
+LL | fn main() { let b: X = Vec::new(); }
+   |                    ^
```

Comment:
Hm, I consider this to be a worse error message than the query cycle, since at least that one's got spans pointing to all the aliases participating in the cycle. But the fixes to overflow handling in weak aliases seems necessary regardless, though, since you could probably have recreated this condition using TAITs.